private/buf/cmd/buf/command/alpha/protoc/protoc.go

```diff
@@ -73,8 +73,10 @@ Additional flags:
 		BindFlags:     flagsBuilder.Bind,
 		NormalizeFlag: flagsBuilder.Normalize,
 		Version: fmt.Sprintf(
-			"%v.%v.%v-buf",
-			bufprotopluginexec.DefaultVersion.GetMajor(),
+			"%v.%v-buf",
+			// DefaultVersion has an extra major version that corresponds to
+			// backwards-compatibility level of C++ runtime. The actual version
+			// of the compiler is just the minor and patch versions.
```

Comment:
I checked historical versions of protoc and everything after 22.0 doesn't use the C++ runtime version. Do we want `bufprotopluginexec.DefaultVersion` to update to match?

```
$ ./protoc-versions.sh                    
libprotoc 3.21.0
libprotoc 3.21.1
libprotoc 3.21.2
libprotoc 3.21.3
libprotoc 3.21.4
libprotoc 3.21.5
libprotoc 3.21.6
libprotoc 3.21.7
libprotoc 3.21.8
libprotoc 3.21.9
libprotoc 3.21.10
libprotoc 3.21.11
libprotoc 3.21.12
libprotoc 22.0
libprotoc 22.1
libprotoc 22.2
libprotoc 22.3
libprotoc 22.4
libprotoc 22.5
libprotoc 23.0
libprotoc 23.1
libprotoc 23.2
libprotoc 23.3
libprotoc 23.4
libprotoc 24.0
libprotoc 24.1
libprotoc 24.2
libprotoc 24.3
libprotoc 24.4
libprotoc 25.0
libprotoc 25.1
libprotoc 25.2
libprotoc 25.3
libprotoc 26.0
libprotoc 26.1
libprotoc 27.0
libprotoc 27.1
```

Reply:
> Do we want bufprotopluginexec.DefaultVersion to update to match?

I don't believe so because that version is also provided to plugins from `buf alpha protoc`, and IIUC in that case `protoc` _does_ include the extra major version number.

Reply:
But I could be wrong. I'm basing that based on the fact that if you change `DefaultVersion` to not include that, you'll get test failures because the extra digit is expected. IIUC, those tests are actually running `protoc` and expecting the output to match that of `buf alpha protoc`. So I believe that means that the major version 5 is passed to plugins in the version number in the code gen request.

Reply:
I *believe* I've seen the major version number change based on the plugin. For example, Java generated code now uses `4.27.1`. C++ and Python use `5.27.1`.

Reply:
Regardless - let's leave this for now in this PR. Resolving.

Reply:
That's kind of funny that 3.21 is reported with the 3 major version since the GitHub release does _not_ have it: it's just [v21.0](https://github.com/protocolbuffers/protobuf/releases/tag/v21.0). I guess they forgot to update the command-line version output until v22. Doh!